[improve][doc] Improve TLS transport encryption and authentication (#16924)
apache/pulsar: site2/docs/security-tls-transport.md

For TLS authentication, the server uses the **trust cert** to verify that the client has a key pair that the certificate authority signed. The common name of the **client cert** is then used as the client's role token (see [Overview](security-overview.md)).

> In Pulsar, we are using [netty-tcnative](https://github.com/netty/netty-tcnative) and [Conscrypt](https://github.com/google/conscrypt) as security provider. We have two format certificates, Java KeyStore(JKS) and CAcerts. When using the JKS, we choose the Conscrypt, this applies to Broker service and Web service. When using the CAcerts on the broker service, we choose the netty-tcnative, which includes two implementations, OpenSSL and JDK, which default to OpenSSL, when the OpenSSL is unavailable, we switch to the JDK.

Reviewer: ```suggestion
Pulsar use [netty-tcnative](https://github.com/netty/netty-tcnative) and [Conscrypt](https://github.com/google/conscrypt) as security providers. There are two certificate formats: 
* Java KeyStore(JKS): When using JKS, you can choose Conscrypt, which applies to both broker service and Web service. 
* CAcerts: When using CAcerts for broker service, you can choose netty-tcnative, which includes two implementations, OpenSSL (default) and JDK. When OpenSSL is unavailable, JDK is used.
```

Author: @momo-jun Thanks! Done.

At this point, you have a cert, `broker.cert.pem`, and a key, `broker.key-pk8.pem`, which you can use along with `ca.cert.pem` to configure TLS transport encryption for your broker and proxy nodes.

### Client certificate

1. Enter the command below to generate the key.

```bash

openssl genrsa -out client.key.pem 2048

```

The client expects the key to be in [PKCS 8](https://en.wikipedia.org/wiki/PKCS_8) format, so enter the following command to convert it.

```bash

openssl pkcs8 -topk8 -inform PEM -outform PEM \
      -in client.key.pem -out client.key-pk8.pem -nocrypt

```

2. Enter the following command to generate the certificate request.

```bash

openssl req -config openssl.cnf \
    -key client.key.pem -new -sha256 -out client.csr.pem

```

3. Sign it with the certificate authority by entering the command below.

```bash

openssl ca -config openssl.cnf -extensions client_cert \
    -days 1000 -notext -md sha256 \
    -in client.csr.pem -out client.cert.pem

```

> At this point, you have a cert, `client.cert.pem`, and a key, `client.key-pk8.pem`, which you can use along with `ca.cert.pem` to configure TLS transport encryption for your client.

Reviewer: ```suggestion
At this point, you have a cert `client.cert.pem` and a key `client.key-pk8.pem`, which you can use along with `ca.cert.pem` to configure TLS encryption for your client.
```
